### PR TITLE
Feature/initial version of the idp client

### PR DESCRIPTION
**Pull request by ivaylomitrev**
I suggest that the PR is reviewed commit by commit. The first commit contains the unchanged code copied from our internal repository. This is so that you can see what changes I have made to it in subsequent commits.

I intend to squash all commits into a single one before merging.

I am still unsure if there is anything special about submitting multi-module Maven projects to Maven central considering that this build will produce multiple POM, JAR, Source, and JavaDoc files, but I will research that further. The same applies to the task we need to create for Sonatype.

@douzounov , I have added you FYI only.

### CODE REVIEW

File: documaster-ws-client-idp/pom.xml [hunk 1]
```diff
@@ -11,6 +11,7 @@
 	</parent>
 
 	<artifactId>documaster-ws-client-idp</artifactId>
+	<name>Documaster Noark 5 identity provider client</name>
```
Comment (teodoradobreva) [inline documaster-ws-client-idp/pom.xml]: We can add description to the child modules as well as the parent one will be inherited.

Comment (ivaylomitrev) [reply documaster-ws-client-idp/pom.xml]: Added.

**Review by teodoradobreva — APPROVED** [review]

**Comment by ivaylomitrev** [comment]
Squashed all commits in preparation for merging.

**Comment by ivaylomitrev** [comment]
Changed the base branch from master to develop.

**Pull request merged**